src/OrleansRuntime/Core/Dispatcher.cs

```diff
@@ -145,27 +145,26 @@ public void ReceiveMessage(Message message)
                         // we want to make sure that we don't un-register the activation we just created.
                         // We would add a counter here, except that there's already a counter for this in the Catalog.
                         // Note that this has to run in a non-null scheduler context, so we always queue it to the catalog's context
-                        if (config.Globals.DirectoryLazyDeregistrationDelay > TimeSpan.Zero)
```

Comment:
With GSI, I have additional situations in which I need to execute this code (specifically, to remove directory entries that point to an activation in a different cluster). The condition is still being checked before the entry is actually removed (in `GrainDirectoryPartition.OkToRemove()').

But, since we are taking about this:  I don't think we should allow users to set DirectoryLazyDeregistrationDelay=0 at all. It makes the system behave incorrectly (specifically, registrations are never cleaned up if the first deregistration fails). Why would you ever want that?